src/mapblock_mesh.cpp

```diff
@@ -618,51 +605,46 @@ static u8 face_contents(content_t m1, content_t m2, bool *equivalent,
 {
 	*equivalent = false;
 
-	if(m1 == CONTENT_IGNORE || m2 == CONTENT_IGNORE)
+	if (m1 == CONTENT_IGNORE || m2 == CONTENT_IGNORE || m1 == m2)
 		return 0;
 
-	bool contents_differ = (m1 != m2);
-
 	const ContentFeatures &f1 = ndef->get(m1);
 	const ContentFeatures &f2 = ndef->get(m2);
 
 	// Contents don't differ for different forms of same liquid
-	if(f1.sameLiquid(f2))
-		contents_differ = false;
+	if (f1.sameLiquid(f2))
+		return 0;
 
 	u8 c1 = f1.solidness;
 	u8 c2 = f2.solidness;
 
-	bool solidness_differs = (c1 != c2);
-	bool makes_face = contents_differ && solidness_differs;
-
-	if(makes_face == false)
+	if (c1 == c2)
 		return 0;
 
-	if(c1 == 0)
+	if (c1 == 0)
 		c1 = f1.visual_solidness;
-	if(c2 == 0)
+	else if (c2 == 0)
 		c2 = f2.visual_solidness;
 
-	if(c1 == c2){
+	if (c1 == c2) {
 		*equivalent = true;
 		// If same solidness, liquid takes precense
-		if(f1.isLiquid())
+		if (f1.isLiquid())
 			return 1;
-		if(f2.isLiquid())
+		if (f2.isLiquid())
 			return 2;
 	}
 
-	if(c1 > c2)
+	if (c1 > c2)
 		return 1;
-	else
-		return 2;
+
+	return 2;
 }
 
 /*
 	Gets nth node tile (0 <= n <= 5).
 */
-TileSpec getNodeTileN(MapNode mn, v3s16 p, u8 tileindex, MeshMakeData *data)
+inline TileSpec getNodeTileN(MapNode mn, v3s16 p, u8 tileindex, MeshMakeData *data)
```

Comment:
llrc, inline makes the function only accessible inside the file, l didn't get an undefined reference error because getNodeTileN is only used in mapblock_mesh.cpp l assume.